catalog/solr/catalog-solr-offline-gazetteer/src/test/groovy/ddf/catalog/solr/offlinegazetteer/GazetteerQueryOfflineSolrSpec.groovy

```diff
@@ -0,0 +1,402 @@
+import spock.lang.Specification
+
+import java.util.concurrent.ExecutorService
+import java.util.stream.Stream
+
+class GazetteerQueryOfflineSolrSpec extends Specification {
+    GazetteerQueryOfflineSolr testedClass
+    SolrClientFactory solrClientFactory
+    SolrClient solrClient
+    ExecutorService executorService
+
+    void setup() {
+        executorService = Mock(ExecutorService)
+        solrClient = Mock(SolrClient)
+        solrClientFactory = Mock(SolrClientFactory) {
+            newClient(_) >> solrClient
+        }
+        testedClass = new GazetteerQueryOfflineSolr(solrClientFactory, executorService)
+    }
+
+    def "Test normal startup ping and suggester build"() {
+        setup:
+        executorService = Mock(ExecutorService) {
+            submit(_ as Runnable) >> {Runnable task -> task.run()}
+        }
+        solrClient = Mock(SolrClient)
+        solrClientFactory = Mock(SolrClientFactory) {
+            newClient(_) >> solrClient
+        }
+
+        2 * solrClient.ping() >> { throw new SolrServerException("exception") } >> Mock(
+                SolrPingResponse)
+
+        1 * solrClient.query(*_) >> { SolrQuery query , METHOD method ->
+            assert query.requestHandler == "/suggest"
+            assert query.get("suggest.build") == "true"
+            assert query.get("suggest.dictionary") == "suggestPlace"
+        }
+
+        when:
+        testedClass = new GazetteerQueryOfflineSolr(solrClientFactory, executorService)
+
+        then:
+        notThrown(Exception)
+    }
+
+    def "Test normal query"() {
+        setup:
+        int numResults = 10
+        1 * solrClient.query({ SolrQuery it -> it.getRows() == numResults }, *_) >>
+                Mock(QueryResponse) {
+                    getResults() >> Mock(SolrDocumentList) {
+                        stream() >> {
+                            Stream.of(Mock(SolrDocument) {
+                                get("title_txt") >> ["title"]
+                                get("ext.population_lng") >> [1337l]
+                                get("location_geo") >> ["POINT (-98.86253 29.18968)"]
+                                get("ext.feature-code_txt") >> ["PPL"]
+                                get("location.country-code_txt") >> ["USA"]
+                                get("ext.gazetteer-sort-value_int") >> [42i]
+                            })
+                        }
+                    }
+                }
+
+        when:
+        List<GeoEntry> results = testedClass.query("sample", numResults)
+
+        then:
+        results.size() == 1
+        with(results.first()) {
+            name == "title"
+            population == 1337
+            29.18 <= latitude && latitude <= 29.19
+            -98.9 <= longitude && longitude <= -98.8
+            featureCode == "PPL"
+            countryCode == "USA"
+        }
+
+    }
+
+    def "Test query max results"() {
+        setup:
+        int numResults = 101
```

Comment:
❓  This is just one higher than max results right? So our max should kick in and correct the issue, only returning `100` results instead?

Reply:
yes, only 100 are allowed so it'll will cap it at 100.